js/script.js

```diff
@@ -608,3 +609,11 @@ function encode(json) {
     var bstring = "0" + base64;
     return bstring;
 }
+
+function factorioEntityObjectArrayFromMultiArray(sourceArray){
+    var objectArray = new Array();
+    for(var i = 0; i < sourceArray.length; i++){
+        objectArray.push(new Entity(sourceArray[i][0],sourceArray[i][1],sourceArray[i][2],sourceArray[i][3],sourceArray[i][4],sourceArray[i][5]));

```

Comment:
can you use intermediate variable names here instead of directly passing each property. It's hard to understand what values are being passed around here.

Eg. 
```javascript
var entityData = sourceArray[i];
var name = entityData[0];
var rotation = entityData[1];
// etc...
```